src/main.tsx

```diff
@@ -20,27 +20,20 @@ import PlaygroundView from "./features/playground/playgroundView";
 import ConfigTableExampleView from "./features/configTable/configTableExample";
 import MovieTableView from "./features/paginatedTable/moviePage";
 import VideoGamesTableView from "./features/formExample/videoGamesPage";
+import { mainSections } from "./siteMap/mainSections";
+import { useEffect } from "react";
 
 const Component = () => {
   return (
     <>
       <Routes>
         <Route index path="/" element={<Navigate to="/test" />} />
         <Route path="/Unauthorized" element={<Unauthorised />} />
-        <Route path="/jsonplaceholder" element={<JsonPlaceHolderView />} />
-        <Route path="/playground" element={<PlaygroundView />} />
-        <Route path="/configTable" element={<ConfigTableExampleView />} />
-        <Route path="/moviesTable" element={<MovieTableView />} />
-        <Route path="/videoGamesTable" element={<VideoGamesTableView />} />
-
-        {
-          //Understand why with this method :params in path are not being recognized.
-          /*mainSections.map((x) =>
-          x.subsections.map((s) => (
-            <Route path={s.path} element={s.component({ a: dispatch })} />
-          ))
-        )*/
-        }
+        {mainSections.map((x) =>
+          x.subsections.map((s) =>
+            s.component ? <Route path={s.path} element={s.component()} /> : null
```

Comment:
the mainSections is a nested structure.

each mainSection has a path (x.path) and nested each subsection. 

you need a <Route> for each main section, and another Router for each subsection inside

I expect a route /main/jsonplaceholder

Reply:
it is already a double map, we only need routes for subsections in this design. The main section is only to group them in the accordion in the SideBar. I am removing the path in the main sections